pkg/sql/schemachanger/scexec/exec_mutation.go

```diff
@@ -118,8 +120,20 @@ func executeDescriptorMutationOps(ctx context.Context, deps Dependencies, ops []
 		}
 	}
 
-	if err := deps.EventLogger().ProcessAndSubmitEvents(ctx); err != nil {
-		return err
+	statementIDs := make([]uint32, 0, len(mvs.eventsByStatement))
+	for statementID := range mvs.eventsByStatement {
+		statementIDs = append(statementIDs, statementID)
+	}
+	sort.Slice(statementIDs, func(i, j int) bool {
+		return statementIDs[i] < statementIDs[j]
+	})
+	for _, statementID := range statementIDs {
+		entries := eventLogEntriesForStatement(mvs.eventsByStatement[statementID])
+		for _, e := range entries {
+			if err := deps.EventLogger().LogEvent(ctx, e.id, *e.metadata, e.event); err != nil {
```

Comment:
It'd be nice to make this interface and call batched. We're pretty close to having that all line up. Happy to defer to a later PR. Thoughts on a TODO here or on the interface?

Reply:
Good call. I'll do what I can.